modules/account-lib/src/coin/dot/walletInitializationBuilder.ts

```diff
@@ -1,26 +1,143 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
-import { Transaction } from './transaction';
+import { decode, methods } from '@substrate/txwrapper-polkadot';
+import BigNumber from 'bignumber.js';
+import { InvalidTransactionError } from '../baseCoin/errors';
 import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TransactionType } from '../baseCoin';
+import { AddProxyArgs, MethodNames, proxyType } from './iface';
+import { WalletInitializationSchema } from './txnSchema';
+import { BaseAddress } from '../baseCoin/iface';
 
 export class WalletInitializationBuilder extends TransactionBuilder {
+  protected _delegate: string;
+  protected _proxyType: proxyType;
+  protected _delay: string;
+
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
+  /**
+   * Register a proxy account for the sender that is able to make calls on its behalf.
+   *
+   * @returns {UnsignedTransaction} an unsigned Dot transaction
+   *
+   * @see https://polkadot.js.org/docs/substrate/extrinsics/#proxy
+   */
+  protected buildTransaction(): UnsignedTransaction {
+    const baseTxInfo = this.createBaseTxInfo();
+    return methods.proxy.addProxy(
+      {
+        delegate: this._delegate,
+        proxyType: this._proxyType,
+        delay: this._delay,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.WalletInitialization;
+  }
+
+  /**
+   * The account to delegate auth to.
+   *
+   * @param {BaseAddress} owner
+   * @returns {WalletInitializationBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#why-use-a-proxy
+   */
+  owner(owner: BaseAddress): this {
+    this.validateAddress({ address: owner.address });
+    this._delegate = owner.address;
+    return this;
+  }
+
+  /**
+   * The proxy type to add.
+   *
+   * @param {proxyType} proxyType
+   * @returns {WalletInitializationBuilder} This builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#proxy-types
+   */
+  type(proxyType: proxyType): this {
+    this._proxyType = proxyType;
+    return this;
+  }
+
+  /**
+   * The number of blocks that an announcement must be in place for.
+   * before the corresponding call may be dispatched.
+   * If zero, then no announcement is needed.
+   *
+   * @param {string} delay
+   * @returns {WalletInitializationBuilder} This transfer builder.
+   *
+   * @see https://wiki.polkadot.network/docs/learn-proxies#time-delayed-proxies
+   */
+  delay(delay: string): this {
```

Comment:
This looks like it should go into the `validityWindow`method but I'm not sure how. Can you add a `TODO: move to the validity window method once it has been standardized`?